File: Solutions/1721. Swapping Nodes in a Linked List/1721. Swapping Nodes in a Linked List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* swapNodes(ListNode* head, int k) {
        ListNode* p1=nullptr,*p2=nullptr;
        for(auto p=head;p;p=p->next){
            p2=p2?p2->next:nullptr;
            if(--k==0){
                p1=p;
                p2=head;
            }
        }
        swap(p1->val,p2->val);
        return head;
    }
};

File: Solutions/1721. Swapping Nodes in a Linked List/1721. Swapping Nodes in a Linked List.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func swapNodes(head *ListNode, k int) *ListNode {
    p1,p2:=new(ListNode),new(ListNode)
    for p:=head;p!=nil;p=p.Next{
        if p2==nil{
            p2=nil
        }else{
            p2=p2.Next
        }
        k-=1
        if k==0{
            p1=p
            p2=head
        }
    }
    p1.Val,p2.Val=p2.Val,p1.Val
    return head
}